src/MatRing.jl

```diff
@@ -79,16 +79,23 @@ is_finite(R::MatRing) = iszero(nrows(a)) || is_finite(base_ring(R))
 ###############################################################################
 
 @doc raw"""
-    similar(x::Generic.MatrixElem, R::NCRing=base_ring(x))
     similar(x::Generic.MatrixElem, R::NCRing, r::Int, c::Int)
+    similar(x::Generic.MatrixElem, R::NCRing)
     similar(x::Generic.MatrixElem, r::Int, c::Int)
+    similar(x::Generic.MatrixElem)
```

Comment:
TODO: untangle the docstrings for `similar`, `zero`, etc. for "generic" AA matrices; and for matrix ring elements